packages/@coorpacademy-components/src/molecule/questions/free-text/index.js

```diff
@@ -1,27 +1,54 @@
-import React, {useMemo} from 'react';
+import React, {useCallback, useMemo, useState} from 'react';
 import PropTypes from 'prop-types';
-import {noop, get} from 'lodash/fp';
+import {noop, getOr} from 'lodash/fp';
+import {convert} from 'css-color-function';
 import Provider from '../../../atom/provider';
 import style from './style.css';
 
-const FreeText = (props, {skin}) => {
-  const {placeholder = '', value = '', onChange = noop} = props;
+const FreeText = (props, context) => {
+  const {placeholder = '', value = '', onChange = noop, 'aria-label': ariaLabel} = props;
+  const [hovered, setHovered] = useState(false);
 
-  const handleChange = useMemo(() => e => onChange(e.target.value), [onChange]);
-  const skinView = value ? get('common.primary', skin) : null;
+  const handleChange = useCallback(e => onChange(e.target.value), [onChange]);
+  const primarySkinColor = getOr('#00B0FF', 'skin.common.primary', context);
+
+  const handleMouseOver = useCallback(() => setHovered(true), [setHovered]);
+
+  const handleMouseLeave = useCallback(() => setHovered(false), [setHovered]);
+
+  const shadowBoxPrimaryColor = useMemo(() => convert(`color(${primarySkinColor} a(12%))`), [
+    primarySkinColor
+  ]);
 
   return (
-    <div data-name="freeText" className={style.wrapper}>
+    <div
+      data-name="freeText"
+      className={style.wrapper}
+      onMouseOver={handleMouseOver}
+      onMouseLeave={handleMouseLeave}
```

Comment:
pas moyen de gérer sans onMouseOver/Leave le cas: value/not value + hovered/not hovered à partir du css pour 3 couleurs de shadow-box (3 cas de couleur shadow-box dont 2 dynamiques - en fonction de la skin)